content_handler/app.cc

```diff
@@ -70,6 +78,42 @@ App::~App() {
   icu_data::Release();
   blink::Threads::Gpu()->PostTask(QuitMessageLoop);
   blink::Threads::IO()->PostTask(QuitMessageLoop);
+  g_app = nullptr;
+}
+
+App& App::Shared() {
+  FTL_DCHECK(g_app);
+  return *g_app;
+}
+
+void App::WaitForPlatformViewIds(
+    std::vector<PlatformViewInfo>* platform_view_ids) {
+  ftl::AutoResetWaitableEvent latch;
+
+  blink::Threads::UI()->PostTask([this, platform_view_ids, &latch]() {
+    WaitForPlatformViewsIdsUIThread(platform_view_ids, &latch);
+  });
+
+  latch.Wait();
+}
+
+void App::WaitForPlatformViewsIdsUIThread(
+    std::vector<PlatformViewInfo>* platform_view_ids,
+    ftl::AutoResetWaitableEvent* latch) {
+  for (auto it = controllers_.begin(); it != controllers_.end(); it++) {
+    ApplicationControllerImpl* controller = it->first;
+
+    if (!controller) {
+      continue;
+    }
+
+    PlatformViewInfo info;
+    info.view_id = reinterpret_cast<uintptr_t>(controller);
```

Comment:
We should create real IDs for these objects instead of relying upon the address of the controller.  The controller's address can be reused, which could confuse us later on.